src/handlers/ResponseHandlers.ts

```diff
@@ -38,6 +39,43 @@ const error: ResponseHandler = {
     },
 };
 
+const blockedByWAF: ResponseHandler = {
+    canProcess: (response) => response.headers.get('x-coveo-waf-action') === 'block',
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'Request blocked for security reasons';
+        platformError.detail = 'The web application firewall has identified the request to be potentially malicious.';
+        platformError.status = response.status;
+        throw platformError;
+    },
+};
+
+const badGateway: ResponseHandler = {
+    canProcess: (response) => response.status === 502,
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'Endpoint unreachable';
+        platformError.detail =
+            'The service is currently unable to reach the necessary endpoint, likely due to a bad gateway. Please try your request again later.';
+        platformError.status = response.status;
+        throw platformError;
+    },
+};
+
+const htmlError: ResponseHandler = {
+    canProcess: (response) => response.headers.get('content-type') === 'text/html',
+    process: async (response) => {
+        const platformError = new CoveoPlatformClientError();
+        platformError.xRequestId = response.headers.get('X-Request-ID') ?? 'unknown';
+        platformError.title = 'There is some issue with the endpoint and we cant parse the error'; // TODO: technical writer
```

Comment:
@rtherien Thanks for the revision, I updated my pr, there is one last string that needs a revision